src/fs/file.rs

```diff
@@ -84,6 +97,11 @@ pub struct File<'dir> {
 
     /// The absolute value of this path, used to look up mount points.
     absolute_path: OnceLock<Option<PathBuf>>,
+
+    pub total_size: bool,
+
+    /// The recursive directory size when total_size is used.
+    recursive_size: Option<u64>,
```

Comment:
Reviewing my own code.

`total_size` and `recursive_size` can be combined into a new enum